dev/puppet-server.conf.sample

```diff
@@ -30,8 +30,11 @@ webserver: {
 # configure the mount points for the web apps
 web-router-service: {
     "puppetlabs.services.ca.certificate-authority-service/certificate-authority-service": "/puppet-ca"
-    "puppetlabs.services.master.master-service/master-service": "/puppet"
+    "puppetlabs.services.master.master-service/master-service": {master-routes:       "/puppet"
 
+                                                                 # The following controls 404 errors and should not
+                                                                 # be modified
+                                                                 invalid-in-puppet-4: "/" }
```

Comment:
Could you refer me to the comment and Chris addressing it?  I'm having a heck of a time finding it.

To be clear, I'm not trying to prescribe how we're doing it.  I'd expect the solution would involve checking something other than the URL itself, because the URI alone doesn't give a lot of indication that the request was previously valid, but is now invalid.

I'd expect the solution to involve checking the request headers, or some other meta-data in addition to the URI to figure out if the request is a from an agent that needs to be updated.